pkg/restapi/issuer/operation/operations.go

```diff
@@ -1176,22 +1193,68 @@ func (o *Operation) storeUserInvitationMapping(mapping *UserInvitationMapping) e
 }
 
 // TODO fulfillment shouldn't be saved, need re-design [Issue#560]
-func (o *Operation) saveCredentialFulfillment(thID string, fulfillment *verifiable.Presentation) error {
-	fbytes, err := fulfillment.MarshalJSON()
+func (o *Operation) saveCredentialAttachmentData(thID string, attachments credentialAttachments) error {
+	manifestBytes, err := json.Marshal(attachments.ManifestID)
 	if err != nil {
-		return fmt.Errorf("failed to marshal credential fulfillment: %w", err)
+		return fmt.Errorf("failed to marshal credential manifest: %w", err)
 	}
 
-	err = o.txnStore.Put(thID, fbytes)
+	fulfilmentBytes, err := json.Marshal(attachments.FulFillment)
+	if err != nil {
+		return fmt.Errorf("failed to marshal credential fulfilment: %w", err)
+	}
+
+	err = o.txnStore.Batch([]storage.Operation{
+		{
+			Key:   manifestDataPrefix + thID,
+			Value: manifestBytes,
+		},
+		{
+			Key:   fulfillmentDataPrefix + thID,
+			Value: fulfilmentBytes,
+		},
+	})
 	if err != nil {
-		return fmt.Errorf("failed to save credential fulfillment in store: %w", err)
+		return fmt.Errorf("failed to save credential attachments in store: %w", err)
+	}
+
+	return nil
+}
+
+func (o *Operation) readAndValidateCredentialApplication(msg service.DIDCommAction,
+	credManifest *cm.CredentialManifest) error {
+	// reading credential application if issuer have sent, out presentation_definition along with manifest
+	// TODO [Issue#563] validate signatures and proofs of credential application
+	if credManifest.PresentationDefinition != nil {
+		applicationAttachments, err := getAttachments(msg)
+		if err != nil {
+			return fmt.Errorf("failed to get request credential attachments: %w", err)
+		}
+
+		if len(applicationAttachments) != 1 {
+			return errors.New("waci request credential: expected only one attachment for request credential")
```

Comment:
just rename this message to standardize error message, something like `invalid request credential message, expected valid credential application`